### ISSUE TITLE
[META] Client Side Encryption in OpenSearch 

### ISSUE THREAD

**Issue by vikasvb90** [opensearch-project/OpenSearch#7229]
We have discussed about feature proposal of enhancement of repository plugin in https://github.com/opensearch-project/OpenSearch/issues/6353 and design proposal for the same in https://github.com/opensearch-project/OpenSearch/issues/6633. Based on this we have come up with the following tasks : 
1. Define a new plugin interface for crypto plugin and management of crypto clients and their metadata.
2. Changes in remote store to provide encryption or decryption of content.
3. Common extensible plugin implementation to provide support for crypto abstractions.

This issue is a meta-issue for tracking the progress of client side encryption.
- [ ] ~~#7284~~
- [ ] ~~#7337~~
- [x] #8465
- [x] #8466
- [x] #9289
- [x] #9833
- [x] #9823 (Will be merged after approval from ESDK team on custom changes)
- [x]  #9812 (Deprecated)

API Changes in OpenSearch

```
PUT _snapshot/vikasvb-repository-test-1
{
  "type": "s3",
  "settings": {
    "bucket": "vikasvb-repository-test",
    "region": "us-west-2"
  },    
  "encrypted": true,
  "crypto_settings": {
      "key_provider_name": "first-mock-1",
      "key_provider_type": "sample-key-provider-extension",
      "settings": {
          "key-1": "sample value"
      }
  }
}
```

**Comment by anasalkouz** [comment]
@vikasvb90 is this tracking green for 2.9?

**Comment by vikasvb90** [comment]
@anasalkouz Reviews couldn't start on this so it is not being planned for 2.9 anymore.

**Comment by anasalkouz** [comment]
Moving this to 2.10

**Comment by hdhalter** [comment]
Is this going out in 2.10? If so, can someone please review the PR for technical accuracy? Thanks. https://github.com/opensearch-project/documentation-website/pull/4389

**Comment by vikasvb90** [comment]
> Is this going out in 2.10? If so, can someone please review the PR for technical accuracy? Thanks. https://github.com/opensearch-project/documentation-website/pull/4389 @vikasvb90 , @anasalkouz 

@hdhalter Yes, this is going in 2.10. I will take a look. Thanks!

**Comment by leo-scalingo** [comment]
Hi @vikasvb90 has it landed in 2.10? I can't see a trace of it, if not, what's the state of this change?

**Comment by asifbashar** [comment]
@vikasvb90 what is the plan for this feature when it will become part of opensearch release.
